v5.0.0-beta.2 (#2675)
mui-org/material-ui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 5.0.0-beta.2

_Sep 23, 2021_

A big thanks to the 5 contributors who made this release possible. Here are some highlights ✨:

- 🇻🇳 Add Vietnamese (viVN) locale (#2668) @tuananh281098
- 🇵🇱 Improve Polish (plPL) locale (#2632) @michallukowski

> - ⚡️ Apply the `valueFormatter` to `singleSelect` select options (#2581) @DanailH

Reviewer: It's a bug fix, right?

```suggestion
- 🐛 Fix usage of `valueFormatter` with the `singleSelect` column type (#2581) @DanailH
```

Reviewer 2: It was not applied at all before

Author: It's more of a feature.

Author: ```suggestion
- ⚡️ Apply the `valueFormatter` to the `singleSelect` column type (#2581) @DanailH
```

### `@mui/x-data-grid@v5.0.0-beta.2` / `@mui/x-data-grid-pro@v5.0.0-beta.2`

#### Breaking changes

- [DataGrid] The params passed to the `valueFormatter` were changed. (#2581) @DanailH

>   Use the `api` to get the missing params.

Reviewer: To fix the formatting: https://github.com/mui-org/material-ui-x/blob/65a0671b2983a63d6107cebf7e0cb02093a05ed9/CHANGELOG.md

```suggestion

  Use the `api` to get the missing params.
```

  The `GridValueFormatterParams` interface has the following signature now:

  ```diff
  -export type GridValueFormatterParams = Omit<GridRenderCellParams, 'formattedValue' | 'isEditable'>;
  +export interface GridValueFormatterParams {
  +  /**
  +   * The column field of the cell that triggered the event
  +   */
  +  field: string;
  +  /**
  +   * The cell value, but if the column has valueGetter, use getValue.
  +   */
  +  value: GridCellValue;
  +  /**
  +   * GridApi that let you manipulate the grid.
  +   */
  +  api: any;
  +}

>   ```

Reviewer: Would it make sense to revert this breaking change? It's a beta release, we have exited the alpha state, it didn't seem needed to fix #2514.

#### Changes

- [DataGrid] Add Vietnamese (viVN) locale (#2668) @tuananh281098
- [DataGrid] Apply the `valueFormatter` to `singleSelect` select options (#2581) @DanailH
- [DataGrid] Free up column header space when icons are not visible (#2606) @DanailH
- [DataGrid] Improve Polish (plPL) locale (#2632) @michallukowski

### Docs

- [docs] Add section for controlled selection and server-side pagination (#2602) @DanailH
- [docs] Fix Algolia search (#2655) @oliviertassinari

> - [docs] Fix the seach results to make them helpful (#2656) @oliviertassinari

Reviewer 2: ```suggestion
- [docs] Improve the seach results relevance (#2656) @oliviertassinari
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 5.0.0-beta.2

> _Sep 23, 2021_

Author: ```suggestion
_Sep 24, 2021_
```

### `@mui/x-data-grid@v5.0.0-beta.1` / `@mui/x-data-grid-pro@v5.0.0-beta.1`

> There are no changes since the last release.
> Refer to `v5.0.0-beta.0` for the breaking changes.

Reviewer: Would it be clearer?

```suggestion
This is a hot fix, to fix an important regression with 5.0.0-beta.0.
```
